src/services/upload/exifService.ts

```diff
@@ -1,90 +1,54 @@
-import EXIF from 'exif-js';
+import exifr from 'exifr';
 
 import { logError } from 'utils/sentry';
 import { NULL_LOCATION, Location } from './metadataService';
 
-const SOUTH_DIRECTION = 'S';
-const WEST_DIRECTION = 'W';
-const CHUNK_SIZE_FOR_EXIF_READING = 4 * 1024 * 1024;
+const EXIF_TAGS_NEEDED = [
+    'DateTimeOriginal',
+    'CreateDate',
+    'ModifyDate',
+    'GPSLatitude',
+    'GPSLongitude',
+    'GPSLatitudeRef',
+    'GPSLongitudeRef',
+];
 interface ParsedEXIFData {
     location: Location;
     creationTime: number;
 }
 
 export async function getExifData(
-    worker,
     receivedFile: globalThis.File
 ): Promise<ParsedEXIFData> {
     try {
-        const fileChunk = await worker.getUint8ArrayView(
-            receivedFile.slice(0, CHUNK_SIZE_FOR_EXIF_READING)
-        );
-        const exifData = EXIF.readFromBinaryFile(fileChunk.buffer);
+        const exifData = await exifr.parse(receivedFile, EXIF_TAGS_NEEDED);
         if (!exifData) {
             return { location: NULL_LOCATION, creationTime: null };
         }
-        return {
+        const parsedEXIFData = {
             location: getEXIFLocation(exifData),
             creationTime: getUNIXTime(exifData),
         };
+        return parsedEXIFData;
     } catch (e) {
         logError(e, 'error reading exif data');
         // ignore exif parsing errors
     }
 }
 
 function getUNIXTime(exifData: any) {
-    const dateString: string = exifData.DateTimeOriginal || exifData.DateTime;
-    if (!dateString || dateString === '0000:00:00 00:00:00') {
+    const dateTime =
+        exifData.DateTimeOriginal ?? exifData.CreateDate ?? exifData.ModifyDate;
+
+    if (!dateTime) {
         return null;
     }
-    const parts = dateString.split(' ')[0].split(':');
-    const date = new Date(
-        Number(parts[0]),
-        Number(parts[1]) - 1,
-        Number(parts[2])
-    );
-    return date.getTime() * 1000;
+    return dateTime.getTime() * 1000;
 }
 
 function getEXIFLocation(exifData): Location {
-    if (!exifData.GPSLatitude) {
+    if (!exifData.latitude || !exifData.longitude) {
```

Comment:
the package added this latitude and longitude property by parsing the GPS exif data so custom parsing is not required